### PR TITLE
refactor(editor): SlateOverlay reusable component

### PR DESCRIPTION
**Pull request by hejtful**
Merge overlays from Text plugin link controls, Text plugin suggestions and BlankExercise plugin blank controls into one reusable SlateOverlay component.

Note: The positioning of Text plugin suggestions is slightly different and has a bit lighter shadow.

### CODE REVIEW

File: packages/editor/src/editor-ui/slate-hover-overlay.tsx [hunk 1]
```diff
@@ -1,15 +0,0 @@
-import { useSlate } from 'slate-react'
-
-import { HoverOverlay, type HoverOverlayProps } from './hover-overlay'
```
Comment (elbotho) [inline packages/editor/src/editor-ui/slate-hover-overlay.tsx:3]: pretty sure `HoverOverlay` can then also be removed

File: packages/editor/src/editor-ui/slate-overlay.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      >
+        {children}
+      </div>
+    </div>
+  )
+}
+
+// If provided an anchor element, returns its size and position (DOMRect).
+// Otherwise, checks for native DOM selection, and provides a DOMRect based on it.
+function getAnchorRect(editor: ReactEditor, anchor: CustomElement | undefined) {
```
Comment (elbotho) [inline packages/editor/src/editor-ui/slate-overlay.tsx:56]: could be
```suggestion
function getAnchorRect(editor: ReactEditor, anchor?: CustomElement) {
```

**Review by elbotho — APPROVED** [review]
thank you!

File: packages/editor/src/editor-ui/slate-overlay.tsx (hunk 2 above)
Comment (hejtful) [reply packages/editor/src/editor-ui/slate-overlay.tsx:56]: I did write it like that the first time, but then it looks like that argument can be omitted entirely. In this case, I'd like the function signature to point out that this argument is expected to be passed, but the passed variable can be of value `undefined`. Hope that makes sense?

Comment (elbotho) [reply packages/editor/src/editor-ui/slate-overlay.tsx:56]: > Hope that makes sense?

With your explanation: yes! 
In the wild I'd probably still read it as a more verbose version of `anchor?:…` 
If you want to make it very clear I'd probably go with `CustomElement | null` or something like that.

In this case: I don't mind either way :)

**Comment by elbotho** [comment]
@hejtful merged this one only half intentionally.
Okay for you? :)

**Comment by hejtful** [comment]
Fine with me :)

**Pull request merged**